pkg/trustedresources/sign.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package trustedresources
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	"github.com/sigstore/sigstore/pkg/signature"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+)
+
+// signInterface returns the encoded signature for the given object.
+func signInterface(signer signature.Signer, i interface{}) ([]byte, error) {
```

Comment:
Oh sure! It is due to some historical issues. I planned to add signing cli into pipeline and will call this function. So that's why it has unit tests. But now this signInterface is only used as a helper function in other unit tests.

Reply:
It may be better to move all of them to test/trustedreosurces? There's no reason for them to be in pkg and export signing functions now